lib/urlwatch/handler.py

```diff
@@ -55,6 +57,24 @@ def __init__(self, cache_storage, job):
         self.etag = None
         self.error_ignored = False
 
+    def __enter__(self):
+        try:
+            self.job.request_resources(JobState.resouces)
+        except Exception as ex:
+            self.exception = ex
+            self.traceback = traceback.format_exc()
+        finally:
+            return self
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        try:
+            self.job.release_resources(JobState.resouces)
+        except Exception as ex:
+            logger.error('Got exception while releasing resources for job: %r', self.job, exc_info=ex)
+            if self.exception is None:
+                self.exception = ex
+                self.traceback = traceback.format_exc()
```

Comment:
Error handling here is a little tricky. The job run may be successful during `process()` call, and produced useful results that users would definitely want to see in reports, but we may still want users to be aware of errors happened here. How should we report this?

Reply:
Maybe just a `logger.warn()` or something?

Reply:
I suspect many users are not seeing most logs, because urlwatch would run in the background by a scheduler. But still, this would be preferable to hiding diff results.